GameEngine/GameBoard.cs

```diff
@@ -57,19 +53,39 @@ private BoardPositionType Convert(CardType cardType)
             return BoardPositionType.Nest;
         }
 
+        private int FindOwl(int position)
+        {
+            var owlIndex = OwlPositions.IndexOf(position);
+            if (owlIndex == -1) {
+                throw new InvalidMoveException("There is no owl at position " + position);
+            }
+            return owlIndex;
+        }
+
+        private int FindDestinationPosition(Play play)
+        {
+            var desiredBoardColor = Convert(play.Card);
+            int newPosition = play.Position;
+            Predicate<int> isEmptySpaceOfCorrectColor = (position) =>
```

Comment:
💡 What do you think about extracting this into an instance method?

Reply:
Or maybe keeping it here and adding the `newPosition < NestPosition` check to the `Predicate`...?

Reply:
`FindDestinationPosition` has been refactored. The new version incorporates these suggestions.